.github/workflows/documentation.yaml

```
@@ -0,0 +1,30 @@
+name: Build Documentation
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  coverage:
+    runs-on: ubuntu-24.04
+
+    steps:
+    - name: Checkout code including full history and submodules
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+        fetch-depth: 0
+
+    - name: Install dependencies from APT repository
+      run:  |
+        sudo apt-get update
+        sudo apt-get install doxygen graphviz
+
+    - name: Build Doxygen documentation
+      run: tools/ci/run_ci.sh --run-doxygen
+
+    - name: Upload Doxygen documentation
+      uses: actions/upload-artifact@v4
+      with:
+        name: Doxygen documentation (HTML)
+        path: build-wakaama/doxygen_out
```

```suggestion
        path: build-wakaama/doxygen-out
```

or
```suggestion
        path: build-wakaama/doxygen
```

(see other comment)